Sources/TuistCache/ContentHashing/CoreDataModelsContentHasher.swift

```diff
@@ -0,0 +1,31 @@
+import Foundation
+import TuistCore
+
+public protocol CoreDataModelsContentHashing {
+    func hash(coreDataModels: [CoreDataModel]) throws -> String
+}
+
+public final class CoreDataModelsContentHasher: CoreDataModelsContentHashing {
+    private let contentHasher: ContentHashing
+
+    // MARK: - Init
+
+    public init(contentHasher: ContentHashing = CacheContentHasher()) {
```

Comment:
If you are injecting a new instance of the `CacheContentHasher` in every `_Hasher` class, you are missing the opportunity of caching hashes across all the hashers. I'd recommend passing down the cache as an argument in the hash method instead of the constructor:

```swift
func hash(coreDataModels: [CoreDataModel], contentHasher: CacheContentHashing)
```

Reply:
And the same comment applies to all the places where you are injecting a new instance of `CacheContentHasher` through the constructor.

Reply:
🙈 late night coding mistakes🙈 Nice catch!

I decided to just pass in the same instance of the hasher all around, and deleted the default values in the `init` to avoid this from happening in the future